src/types/from_component.rs

```diff
@@ -173,6 +174,75 @@ impl From<Image> for scene::ImageComponent {
     }
 }
 
+impl TryFrom<Text> for scene::TextComponent {
+    type Error = TypeError;
+
+    fn try_from(text: Text) -> Result<Self, Self::Error> {
+        let style = match text.style {
+            Some(TextStyle::Normal) => scene::TextStyle::Normal,
+            Some(TextStyle::Italic) => scene::TextStyle::Italic,
+            Some(TextStyle::Oblique) => scene::TextStyle::Oblique,
+            None => scene::TextStyle::Normal,
+        };
+        let wrap = match text.wrap {
+            Some(TextWrapMode::None) => scene::TextWrap::None,
+            Some(TextWrapMode::Word) => scene::TextWrap::Word,
+            Some(TextWrapMode::Glyph) => scene::TextWrap::Glyph,
+            None => scene::TextWrap::None,
+        };
+        let weight = match text.weight {
+            Some(TextWeight::Thin) => scene::TextWeight::Thin,
+            Some(TextWeight::ExtraLight) => scene::TextWeight::ExtraLight,
+            Some(TextWeight::Light) => scene::TextWeight::Light,
+            Some(TextWeight::Normal) => scene::TextWeight::Normal,
+            Some(TextWeight::Medium) => scene::TextWeight::Medium,
+            Some(TextWeight::SemiBold) => scene::TextWeight::SemiBold,
+            Some(TextWeight::Bold) => scene::TextWeight::Bold,
+            Some(TextWeight::ExtraBold) => scene::TextWeight::ExtraBold,
+            Some(TextWeight::Black) => scene::TextWeight::Black,
+            None => scene::TextWeight::Normal,
+        };
+        let dimensions = match (text.width, text.height, text.max_width, text.max_height) {
+            (Some(width), Some(height), _, _) => scene::TextDimensions::Fixed { width, height },
+            (None, Some(_), _, _) => {
+                //error height can only be define if width is
+                return Err(TypeError::new(
+                    "height can only be provided if width is also defined.",
+                ));
+            }
+            (Some(width), None, _max_width, max_height) => scene::TextDimensions::FittedColumn {
+                width,
+                max_height: max_height.unwrap_or(MAX_NODE_RESOLUTION.height as f32),
+            },
```

Comment:
[NIT]
Using `_` and `_max_sth` is inconsistent. I don't care which one you use but make it consistent across those matches